zcash_client_backend/src/data_api.rs

```diff
@@ -315,12 +315,32 @@ impl AccountBalance {
     }
 }
 
+/// The kinds of accounts supported by `zcash_client_backend`.
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+pub enum AccountKind {
```

Comment:
I named this `AccountKind` instead of `AccountType` because `type` is a Rust keyword (so precludes nice method names like `account.type()`), and there's precedent here with things like `std::io::ErrorKind`.